src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateClone = { ...state };
+  const stateCloneHistory = [];
+
+  for (const { type, extraData, keysToRemove } of actions) {
```

Comment:
here – it looks okay, but make sure to use this approach carefully so that it's still readable